nest_elephant_tvb/translation/mpi_translator.py

```diff
@@ -3,192 +3,237 @@
 import time
 import numpy as np
 from mpi4py import MPI
-from threading import Thread, Lock
 from nest_elephant_tvb.translation.science_nest_to_tvb import store_data,analyse_data
 
-lock_status=Lock() # locker for manage the transfer of data from thread
-
 def init(path, param, comm, comm_receiver, comm_sender, loggers):
     '''
-    Initialize the translation with MPI. NOTE: more information will be added with more changes. Currently the multithreaded version is still in use.
+    Initialize the translation with MPI.
+    NOTE: more information will be added with more changes. This is still the very first version!
+    
+    TODO: IMPORTANT!!! MPI communication on NEST and TVB side is somewhat hardcoded.
+        --> NEST sends to and receives from rank 0 on transformer side. This works here, since rank 0 is the receiving rank
+            --> https://github.com/sdiazpier/nest-simulator/blob/ba0c9d637d6dce7bcdd09b526b28d20f815085d4/nestkernel/recording_backend_mpi.cpp
+            --> line 397 ff
+        --> TVB sends and receives from rank 0 on transformer side.
+            --> nest_elephant_tvb/translation/test_file/test_receive_nest_to_tvb.py
+            --> line 32 ff
+            --> Here: Rank 1-x are doing analysis/science and sending to TVB
+            --> For now, hardcoded solution. All places with 'rank 0' are replaced with 'rank 1'
+            --> But as soon as we use more than two MPI ranks, this fails again!!!
     
+    TODO: Renaming! Translator -> transformer, etc.
+    TODO: This is only the Nest to TVB direction. For other direction some generalizing changes will be needed.
+    TODO: Use RichEndPoints for communication encapsulation
+    TODO: Seperate 1)Receive 2)Analysis/Science and 3)Send. See also the many Todos in the code
+    TODO: Make use of / enable MPI parallelism! Solve hardcoded communication protocol first
+    TODO: Major question and idea: 
+            Why all the trouble with 'two parts' if everything is synchronized?
+            We receive and send seperately but still wait for each other.
+            With MPI possible --> ONE while-loop that handles sending and receiving.
+            Send/Recv seperated by ranks obviously (paralellism).
```

Comment:
The separation is important because the simulator doesn't wait to send the data. The translator needs to receive and send in same time.

Reply:
Kim could you either clarify what you mean or else remove the question?

Reply:
Removed the question after discussion and clarification. The proper implementation over the next steps will split the functionalities and communication anyways.